lib/sets.bzl

```diff
@@ -132,7 +151,8 @@ def _difference(a, b):
     A set containing the elements that are in `a` but not in `b`.
   """
   _precondition_only_sets_or_lists(a, b)
-  return depset([e for e in a if e not in b])
+  bset = {e: None for e in shorter}
```

Comment:
shorter isn't define here, looks like you missed the line to figure that out.

Reply:
I don't believe there's an easy way to check for hashability or comparability, since Skylark doesn't have exceptions. You could inspect the types of the topmost elements but without recursion you can't really traverse beyond that. I think the best bet is providing two different functions and letting the user choose the one appropriate for their data types.